loaders/dojo/i18nEval.js

```diff
@@ -15,26 +15,42 @@
  * limitations under the License.
  */
 
- /*
-  * Evaluates a Dojo i18n resource bundle and returns the bundle object
-  */
+/*
+ * Evaluates a Dojo i18n resource bundle and returns the bundle object
+ */
 module.exports = function(bundle) {
 	var result, isAmd;
 	function define(arg1, arg2) {
 		isAmd = true;
+
 		if (!arg2) {
+			// overwriting language files may define an easy object
 			result = arg1;
-		} else {
-			if (arg1.length !== 0) {
-				throw new Error("define dependencies not supported in langauge files!");
-			}
+
+		} else if (arg1.length === 0) {
+
+			// without dependencies we can easily call the empty factory function
 			result = arg2(); // call factory function
+
+		} else if (arg1.length === 2 && arg1[0] === "require" && arg1[1] === "exports") {
+
+			// Special case for typescript generated i18n files
+			// tsc is adding virtual dependencies for require and exports
+			// we need the latter as our result
+			let exp = {};
+			arg2(null, exp); // require must not be
```

Comment:
A couple changes.

1. Stay away from `let`, `const`, etc. in emitted code to avoid issues with older browsers.
1. Instead of passing null for `require` pass a function that will throw an error with meaningful information.

```suggestion
			var exp = {};
			var req = function() { throw new Error("Calling require is not supported in language files!") };
			arg2(req, exp);
```